<a href="https://colab.research.google.com/github/hrishikeshchavan99/Black-Friday-Dataset/blob/master/EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Exploratory Data Analysis for Black Friday Dataset**

In [0]:
import pandas as pd

In [0]:
df = pd.read_csv("/content/drive/My Drive/Colab Notebooks/train.csv")

In [3]:
df.head()

User_ID Product_ID Gender  ... Product_Category_2  Product_Category_3 Purchase
0  1000001  P00069042      F  ...                NaN                 NaN     8370
1  1000001  P00248942      F  ...                6.0                14.0    15200
2  1000001  P00087842      F  ...                NaN                 NaN     1422
3  1000001  P00085442      F  ...               14.0                 NaN     1057
4  1000002  P00285442      M  ...                NaN                 NaN     7969

[5 rows x 12 columns]

In [4]:
df.tail()

User_ID Product_ID  ... Product_Category_3 Purchase
550063  1006033  P00372445  ...                NaN      368
550064  1006035  P00375436  ...                NaN      371
550065  1006036  P00375436  ...                NaN      137
550066  1006038  P00375436  ...                NaN      365
550067  1006039  P00371644  ...                NaN      490

[5 rows x 12 columns]

In [5]:
df.shape

(550068, 12)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550068 entries, 0 to 550067
Data columns (total 12 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   User_ID                     550068 non-null  int64  
 1   Product_ID                  550068 non-null  object 
 2   Gender                      550068 non-null  object 
 3   Age                         550068 non-null  object 
 4   Occupation                  550068 non-null  int64  
 5   City_Category               550068 non-null  object 
 6   Stay_In_Current_City_Years  550068 non-null  object 
 7   Marital_Status              550068 non-null  int64  
 8   Product_Category_1          550068 non-null  int64  
 9   Product_Category_2          376430 non-null  float64
 10  Product_Category_3          166821 non-null  float64
 11  Purchase                    550068 non-null  int64  
dtypes: float64(2), int64(5), object(5)
memory usage: 50.4+ MB


In [7]:
df.describe()

User_ID     Occupation  ...  Product_Category_3       Purchase
count  5.500680e+05  550068.000000  ...       166821.000000  550068.000000
mean   1.003029e+06       8.076707  ...           12.668243    9263.968713
std    1.727592e+03       6.522660  ...            4.125338    5023.065394
min    1.000001e+06       0.000000  ...            3.000000      12.000000
25%    1.001516e+06       2.000000  ...            9.000000    5823.000000
50%    1.003077e+06       7.000000  ...           14.000000    8047.000000
75%    1.004478e+06      14.000000  ...           16.000000   12054.000000
max    1.006040e+06      20.000000  ...           18.000000   23961.000000

[8 rows x 7 columns]

#Removing null/missing values

to check the total number of missing values in each column

In [8]:
df.isnull().sum()

User_ID                            0
Product_ID                         0
Gender                             0
Age                                0
Occupation                         0
City_Category                      0
Stay_In_Current_City_Years         0
Marital_Status                     0
Product_Category_1                 0
Product_Category_2            173638
Product_Category_3            383247
Purchase                           0
dtype: int64

There are total 173638 null values in Product_Category_2 and 383247 null values in Product_Category_3            
To handle missing numerical values

Fill missing values backward or forward<br>
Forward fill:

In [0]:
df = df.fillna(method = 'pad')

Check if all null values removed or not

In [10]:
df.isnull().sum()

User_ID                       0
Product_ID                    0
Gender                        0
Age                           0
Occupation                    0
City_Category                 0
Stay_In_Current_City_Years    0
Marital_Status                0
Product_Category_1            0
Product_Category_2            1
Product_Category_3            1
Purchase                      0
dtype: int64

In [11]:
df[['Product_Category_2', 'Product_Category_3']].head()


Product_Category_2  Product_Category_3
0                 NaN                 NaN
1                 6.0                14.0
2                 6.0                14.0
3                14.0                14.0
4                14.0                14.0

Still one null value left each in product category 2 and 3<br>
So remove it using backward fill

In [0]:
df = df.fillna(method = 'bfill')

In [13]:
df.isnull().sum()

User_ID                       0
Product_ID                    0
Gender                        0
Age                           0
Occupation                    0
City_Category                 0
Stay_In_Current_City_Years    0
Marital_Status                0
Product_Category_1            0
Product_Category_2            0
Product_Category_3            0
Purchase                      0
dtype: int64

So all null values have been removed

Check again with assert that there are no missing values in the dataframe to gain confidence



In [0]:
assert pd.notnull(df).all().all()

As it is not throwing any assertion error, all null values have been removed